models/templateBD.go

```diff
@@ -6,26 +6,47 @@ type TemplateBD struct {
 	Value map[string]interface{} `json:",omitempty"`
 }
 
-func NewTemplateBD(ops, path, name, displayName, layer2Unicast string, intersiteBumTrafficAllow, optimizeWanBandwidth, l2Stretch, l3MCast bool, vrfRef, dhcpLabel map[string]interface{}) *TemplateBD {
+func NewTemplateBD(ops, path, name, displayName, layer2Unicast, unkMcastAct, multiDstPktAct, v6unkMcastAct, vmac string, intersiteBumTrafficAllow, optimizeWanBandwidth, l2Stretch, l3MCast, arpFlood, unicastRouting bool, vrfRef, dhcpLabel map[string]interface{}) *TemplateBD {
 	var bdMap map[string]interface{}
 	bdMap = map[string]interface{}{
 		"name":                     name,
 		"displayName":              displayName,
 		"l2UnknownUnicast":         layer2Unicast,
+		"unkMcastAct":              unkMcastAct,
+		"multiDstPktAct":           multiDstPktAct,
+		"v6unkMcastAct":            v6unkMcastAct,
+		"vmac":                     vmac,
+		"arpFlood":                 arpFlood,
+		"unicastRouting":           unicastRouting,
 		"intersiteBumTrafficAllow": intersiteBumTrafficAllow,
 		"optimizeWanBandwidth":     optimizeWanBandwidth,
 		"l2Stretch":                l2Stretch,
 		"l3MCast":                  l3MCast,
 		"vrfRef":                   vrfRef,
 		"dhcpLabel":                dhcpLabel,
-		"subnets":                  []interface{}{},
 	}
 
 	if bdMap["l2UnknownUnicast"] == "" {
 		bdMap["l2UnknownUnicast"] = "flood"
 	}
 
-	if bdMap["dhcpLabel"] == nil {
+	if bdMap["unkMcastAct"] == "optimized_flooding" {
+		bdMap["unkMcastAct"] = "opt-flood"
+	} else {
+		bdMap["unkMcastAct"] = "flood"
+	}
+
+	if bdMap["multiDstPktAct"] == "flood_in_bd" || bdMap["multiDstPktAct"] == "" {
+		bdMap["multiDstPktAct"] = "bd-flood"
+	}
+
+	if bdMap["v6unkMcastAct"] == "optimized_flooding" {
+		bdMap["v6unkMcastAct"] = "opt-flood"
+	} else {
+		bdMap["v6unkMcastAct"] = "flood"
+	}
+
+	if len(dhcpLabel) == 0 {
```

Comment:
Why change the == nil to len 0?

Reply:
I modified code in the BD provider where dhcpLabel is a map. This checks for an empty map.